csharp/private/rules/resgen.bzl

```diff
@@ -0,0 +1,99 @@
+load(
+    "@d2l_rules_csharp//csharp/private:providers.bzl",
+    "CSharpResource",
+)
+
+# Label of the csproj template for ResX compilation
+_TEMPLATE = "@d2l_rules_csharp//csharp/private:rules/ResGen.csproj"
+
+# When compiling the csproj, it will look for the embedded resources related to
+# the path of the csproj on disk. Since the csproj is written by the bazel, it will
+# be in the rule bin directory. If we want to get the relative path to the
+# ResX files, we need to get to a junction/symlink for the source. 
+#
+# The junction/symlink that we are using is 4 directories up (\bazel-out\<sandbox>\bin\<name>\)
+def _bazel_to_relative_path(path):
+  return "../../../../%s" % (path)
+
+def _csharp_resx_impl(ctx):
+    """_csharp_resx_impl emits actions for compiling a resx file."""
+    if not ctx.attr.out:
+        resource_name = ctx.attr.name
+    else:
+        resource_name = ctx.attr.out
+    
+    csproj = ctx.actions.declare_file("%s.csproj" % (ctx.attr.name))
+    resource = ctx.actions.declare_file("obj/Debug/%s/%s.resources" % (ctx.attr.target_framework, resource_name))
+
+    ctx.actions.expand_template(
+        template = ctx.file._csproj_template,
+        output = csproj,
+        substitutions = {
+            "{TargetFramework}": ctx.attr.target_framework,
+            "{Resx}": _bazel_to_relative_path(ctx.file.src.path),
+            "{ManifestResourceName}": resource_name,
+        },
+    )
+
+    toolchain = ctx.toolchains["@d2l_rules_csharp//csharp/private:toolchain_type"]
+    
+    args = ctx.actions.args()
+    args.add("build")
+    args.add(csproj.path)
+    
+    ctx.actions.run(
+        inputs = [ctx.file.src, csproj],
+        outputs = [resource],
+        executable = toolchain.runtime,
+        arguments = [args],
+        mnemonic = "CompileResX",
+        progress_message = "Compiling resx file to binary",
+        env = {

```

Comment:
Without these 4 environment variables, `dotnet` cli just won't run. @j3parker, do you have any ideas on how I might be able to: 1) not have to specify these 2) specify these for each OS

While they are set to `/root/`, they will fail on the linux & macOS builds (as they wouldn't have permission to create `/root/.dotnet`). Additionally without `DOTNET_CLI_TELEMETRY_OPTOUT=1`, the telemetry message will output a bunch.

Reply:
Getting back into this after inspiration, I think the best option for now would be a wrapper over `dotnet` that ensures the appropriate variables are passed in for each execution. This way, we don't have to worry about the `.dotnet` directory of one rule conflicting with that of another.

I looked into other options, but they were mostly dead ends.